resources/builder.inc.sh

```diff
@@ -669,6 +676,72 @@ builder_check_color() {
   done
 }
 
+#
+# For every build action:target in _builder_chosen_action_targets, add
+# its full internal dependency tree
+#
+_builder_add_chosen_action_target_dependencies() {
+  local action_target e i=0 new_actions=()
+
+  # Iterate through every action specified on command line; we use this loop
+  # style so that any new actions added here will also be iteratively checked
+  while (( $i < ${#_builder_chosen_action_targets[@]} )); do
+    action_target=${_builder_chosen_action_targets[$i]}
+
+    # If we have an internal dependency for the chosen action:target pair, add
+    # it to the list, but only if there is a defined output and that output is
+    # missing
+    if [[ ! -z ${_builder_internal_dep[$action_target]+x} ]]; then
+      local dep_output=${_builder_internal_dep[$action_target]}
+      if [[ ! -z ${_builder_dep_path[$dep_output]+x} ]] &&
+          [[ ! -e "$KEYMAN_ROOT/${_builder_dep_path[$dep_output]}" ]]; then
+        if ! _builder_item_in_array "$dep_output" "${_builder_chosen_action_targets[@]}"; then
+          _builder_chosen_action_targets+=($dep_output)
+          new_actions+=($dep_output)
+        fi
+      fi
+    fi
+    i=$((i + 1))
+  done
+
+  if [[ ${#new_actions[@]} -gt 0 ]]; then
+    echo "Automatically running following required actions with missing outputs:"
+    for e in "${new_actions[@]}"; do
+      echo "* $e"
+    done
+  fi
+}
+
+#
+# If we have described outputs, then we will setup our
+# default internal dependency chain:
+#
+#  configure <- build <- (test,install,publish)
+#
+_builder_define_default_internal_dependencies() {
+  for target in "${_builder_targets[@]}"; do
+    _builder_define_default_internal_deps_for_target "$target"
+  done
+
+  _builder_define_default_internal_deps_for_target ':*'
+}
+
+_builder_define_default_internal_deps_for_target() {
+  local target=$1
+  _builder_define_default_internal_dep "$target" configure build
+  _builder_define_default_internal_dep "$target" build test
+  _builder_define_default_internal_dep "$target" build install
+  _builder_define_default_internal_dep "$target" build publish
```

Comment:
Suggestion, along with a new action:

```
_builder_define_default_internal_dep "$target" build upload-sentry
```

I believe that `publish` is meant to be separate from Sentry symbol uploads, right?  After a fair bit of thought on Web's build script, it seems that an action is most appropriate for the Sentry task - after all, two separate targets for Web (`-embed`, `-web` in old-script form) have a corresponding upload.

Reply:
Alternative form:  `symbolify` (or similar)

Reply:
I am not sure if we need or even want this. The internal dependency pattern is intended to streamline use of the build scripts in interactive use on the command line. For `publish-sentry` or `publish-symbols` or what ever we end up calling it, this is a part of CI where we are always explicit about what we are building -- we will have just configured, built and tested before running any publish steps.

Reply:
Which also means I should remove line 734 above, because it doesn't make sense for the same reasons. So, probably not the outcome you were expecting but thank you!

Reply:
I based that suggestion on what the removed line (re: `publish`) implied.  If that implication itself is the issue, fine by me.

Reply:
Indeed, fresh day and I saw this from a clearer perspective. Your comment precipitated my change here!